From a8cfbd698705c1244ae2917a7c5156f4c37dacc8 Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Thu, 28 Apr 2011 13:57:06 +0800
Subject: [PATCH 917/998] ENGR00142977 SDHCI Refine the special delay for SMD eMMC

In order to decrease the CPU loading, use CMD6(only called in the
initialization procedure) but not CMD13 as the delay trigger.

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 drivers/mmc/host/mx_sdhci.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/mmc/host/mx_sdhci.c b/drivers/mmc/host/mx_sdhci.c
index 438edcf..b89bb42 100644
--- a/drivers/mmc/host/mx_sdhci.c
+++ b/drivers/mmc/host/mx_sdhci.c
@@ -804,12 +804,6 @@ static void sdhci_send_command(struct sdhci_host *host, struct mmc_command *cmd)
 	/* Configure the cmd type for cmd12 */
 	if (cmd->opcode == 12)
 		mode |= SDHCI_TRNS_ABORTCMD;
-	/*
-	 * Some delay is mandatory required between CMD6 and CMD13 after
-	 * switch to DDR mode when Sandisk eMMC44 soldered on SMD board
-	 */
-	if (cmd->opcode == 0xd)
-		mdelay(5);
 	DBG("Complete sending cmd, transfer mode would be 0x%x.\n", mode);
 	writel(mode, host->ioaddr + SDHCI_TRANSFER_MODE);
 }
@@ -848,6 +842,12 @@ static void sdhci_finish_command(struct sdhci_host *host)
 	if (!host->cmd->data)
 		queue_work(host->workqueue, &host->finish_wq);
 
+	/*
+	 * Some delay is mandatory required between CMD6 and CMD13 after
+	 * switch to DDR mode when Sandisk eMMC44 soldered on SMD board
+	 */
+	if (host->cmd->opcode == 0x6)
+		mdelay(5);
 	host->cmd = NULL;
 }
 
-- 
1.7.4.1

